testdata/e2e/e2e_test.go

```diff
@@ -0,0 +1,326 @@
+package test
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	cadvisor "github.com/google/cadvisor/info/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	statsapi "k8s.io/kubernetes/pkg/kubelet/apis/stats/v1alpha1"
+)
+
+func TestMetricSample(t *testing.T) {
+
+	wd := os.Getenv("WORKING_DIR")
+
+	parsedK8sLists := &ParsedK8sLists{
+		NodeSummaries:          make(map[string]statsapi.Summary),
+		BaselineNodeSummaries:  make(map[string]statsapi.Summary),
+		NodeContainers:         make(map[string]map[string]cadvisor.ContainerInfo),
+		BaselineNodeContainers: make(map[string]map[string]cadvisor.ContainerInfo),
+	}
+
+	t.Run("ensure that a metrics sample has expected files", func(t *testing.T) {
+		seen := make(map[string]bool, len(knownFileTypes))
+
+		err := filepath.Walk(wd, func(path string, info os.FileInfo, e error) error {
+			if e != nil {
+				return e
+			}
+
+			// check if it is a regular file (not dir)
+			if info.Mode().IsRegular() {
+				n := info.Name()
+				ft := toAgentFileType(n)
+				seen[toAgentFileType(ft)] = true
+				if unmarshalFn, ok := knownFileTypes[ft]; ok {
+					fmt.Println("Processing:", n)
+					f, err := ioutil.ReadFile(path)
+					if err != nil {
+						return err
+					}
+
+					if err := unmarshalFn(path, f, parsedK8sLists); err != nil {
+						return err
+					}
+
+				}
+			}
+
+			return nil
+		})
+		if err != nil {
+			t.Fatalf("Failed: %v", err)
+		}
+		err = checkForRequiredFiles(seen)
+		if err != nil {
+			t.Fatalf("Failed: %v", err)
+		}
+	})
+
+	t.Parallel()
+
+	t.Run("ensure that a metrics sample contains the cloudability namespace", func(t *testing.T) {
+		for _, ns := range parsedK8sLists.Namespaces.Items {
+			if ns.Name == "cloudability" {
+				return
+			}
+		}
+		t.Error("Namespace cloudability not found in metric sample")
+	})
+
+	t.Run("ensure that a metrics sample has expected pod data", func(t *testing.T) {
+		for _, po := range parsedK8sLists.Pods.Items {
+			if strings.HasPrefix(po.Name, "stress-") && po.Status.QOSClass == v1.PodQOSBestEffort {
+				return
+			}
+
+		}
+		t.Error("pod stress not found in metric sample")
+	})
+
+	t.Run("ensure that a metrics sample has expected containers summary data", func(t *testing.T) {
+
+		for _, ns := range parsedK8sLists.NodeSummaries {
+			for _, pf := range ns.Pods {
+				if strings.HasPrefix(pf.PodRef.Name, "stress-") && pf.PodRef.Namespace == "stress" && pf.CPU.UsageNanoCores != nil {
+					return
```

Comment:
Does having the return in the nested for loop mean that we will pass if any NodeSummaries has the required data? Did we want to check if all NodeSummaries have the required data?

Reply:
Currently we only provision the kind cluster with one node, but if we move to using multiple nodes the pod will schedule the stress reference workload on only one node. So I feel that this current behavior is correct.